[recipe]: Add `app-monochrome`

appetrosyan opened the pull request:
### Brief summary of what the package does

Low contrast monochromatic theme suitable for OLED that uses font shapes rather than colour for highlighting.

### Direct link to the package repository
https://github.com/Greybeard-Entertainment/app-monochrome

### Your association with the package

Author and maintainer

### Relevant communications with the upstream package maintainer

None

### Checklist


- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

tarsius commented:
A package must feature a library that matches its name. For multi-variant themes, we recommend adding a stub library named `*-themes.el`, which contains nothing but a valid header and a library commentary (which will become the package description on Melpa).

So in this case, the package should be renamed to `app-monochrome-themes` and a library `app-monochrome-themes.el` should be added.

appetrosyan commented:
Got it, thank you!

appetrosyan commented:
Done.

tarsius commented:
You still have to rename the recipe and change the package symbol in it.

riscy reviewed (commented):
We have a WIP FAQ that I haven't touched up enough to make "official" yet, but this may be helpful: https://github.com/melpa/melpa/blob/4caa9a03cad40d24484fce27fcfcf75b1d08e814/FAQ.md#how-do-i-make-a-package-containing-multiple-themes

I've made a suggestion as to how your recipe should appear, and under that recipe the following pops up:

`app-monochrome-dark-theme.el` with package-lint 20240118.1725 and `package-lint-main-file` = "app-monochrome-themes.el":
```
3 issues found:
1:0: error: There is no `provide' form.
6:0: error: Package-Requires outside the main file have no effect.
61:0: error: "app-monochrome-dark" doesn't start with package's prefix "app-monochrome-themes".
```

`app-monochrome-dark-theme.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
```

`app-monochrome-light-theme.el` with package-lint 20240118.1725 and `package-lint-main-file` = "app-monochrome-themes.el":
```
3 issues found:
1:0: error: There is no `provide' form.
6:0: error: Package-Requires outside the main file have no effect.
59:0: error: "app-monochrome-light" doesn't start with package's prefix "app-monochrome-themes".
```

`app-monochrome-light-theme.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
```

`app-monochrome-themes.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
```

Package and license:
- app-monochrome-dark-theme.el -- not in package namespace `app-monochrome-themes-`
- app-monochrome-light-theme.el -- not in package namespace `app-monochrome-themes-`

appetrosyan commented on recipes/app-monochrome-theme:
```diff
@@ -0,0 +1,4 @@
+(app-monochrome-theme
+ :fetcher github
+ :repo "Greybeard-Entertainment/app-monochrome"
+ :files "*.el")
```
Alright, this is all somewhat new to me, so thank you and thanks to @tarsius for your patience.

appetrosyan replied:
@riscy 

I have _fixed_ the problems with the themes to the best of my ability, including adding `melpazoid` to my CI. 

In order to conform to the 80 character limit I have removed the license statements from each font bullet, instead replacing it with an overall statement. This is far less precise but I suppose people with a strict 80 character limit wouldn't see that information anyway.

I am somewhat confused as to what the naming convention should be, so I left the packages the way they are, which is more in line with e.g. [modus themes](https://github.com/protesilaos/modus-themes). 

I hope that this is one case where strict adherence to `melpazoid`'s suggestions is not required for a successful merge. 

---- 
The FAQ has a typo, by the way, the autoload contains a space between the third comment sign and the `###` symbols.

riscy commented:
I think you have not given reviewers permission to modify the code in your PR (otherwise I would fix this myself) but you should additionally ensure that your recipe *filename* is the same as the package name (i.e. 'recipes/app-monochrome-theme' does not match 'app-monochrome-themes')

Here's the latest otherwise:

`app-monochrome-dark-theme.el` with package-lint 20240118.1725 and `package-lint-main-file` = "app-monochrome-themes.el":
```
1 issue found:
60:0: error: "app-monochrome-dark" doesn't start with package's prefix "app-monochrome-themes".
```

`app-monochrome-light-theme.el` with package-lint 20240118.1725 and `package-lint-main-file` = "app-monochrome-themes.el":
```
1 issue found:
58:0: error: "app-monochrome-light" doesn't start with package's prefix "app-monochrome-themes".
```

`app-monochrome-themes.el` with checkdoc 0.6.2 (fix *within reason*):
```
app-monochrome-themes.el:60: First sentence should end with punctuation
```

Package and license:
- app-monochrome-dark-theme.el -- not in package namespace `app-monochrome-themes-`
- app-monochrome-light-theme.el -- not in package namespace `app-monochrome-themes-`

appetrosyan commented:
OK. So I've addressed the naming change. I've added a prefix to all of the themes in the set, and I've added you @riscy as a collaborator on that fork, as there will be other bigger packages.

riscy commented:
> I've added you @riscy as a collaborator on that fork, as there will be other bigger packages.

This is not necessary and I probably won't be able to commit any time to this theme or other bigger packages. :)

> (@ :fetcher github :repo "skeeto/at-el")

Something strange happened here - I think the recipe is clearly incorrect now?

appetrosyan commented:
Yeah, my bad, the wrong recipe got renamed.

riscy commented:
Recipe looks fixed - thank you.  Here's where we're at --

`app-monochrome-themes.el` with checkdoc 0.6.2 (fix *within reason*):
```
app-monochrome-themes.el:60: First sentence should end with punctuation
```

Package and license:
- Avoid specifying `:branch` except in unusual cases.  (MELPA vastly prefers to use the default branch as opposed to a custom branch you have)

appetrosyan commented:
Fixed `checkdoc`, renamed main branch to `main` (`barba` was default, but I was paranoid, now it's standard). 

Rebased.

riscy commented:
Thank you for accommodating - and welcome to MELPA. :)

appetrosyan commented:
@riscy The pleasure is all mine. 

I was not familiar with many of the conventions, so hopefully next few packages will be merged without extra hassle.
